packages/react-canvas-core/src/core-state/AbstractDisplacementState.ts

```diff
@@ -63,6 +63,45 @@ export abstract class AbstractDisplacementState<E extends CanvasEngine = CanvasE
 				}
 			})
 		);
+
+		this.registerAction(
+			new Action({
```

Comment:
can we move the logic inside these events to protected methods inside this class, and then simply call them in each action with the correct clientX and client Y (basically remove all the code duplication)

Reply:
Makes sense. Done!